index.d.ts

```diff
@@ -731,8 +731,8 @@ declare module 'mongoose' {
   interface Model<T, TQueryHelpers = {}, TMethods = {}> extends NodeJS.EventEmitter, AcceptsDiscriminator {
     new(doc?: AnyKeys<T> & AnyObject): EnforceDocument<T, TMethods>;
 
-    aggregate<R = any>(pipeline?: any[]): Aggregate<Array<R>>;
-    aggregate<R = any>(pipeline: any[], cb: Function): Promise<Array<R>>;
+    aggregate<R = any>(pipeline?: any[], options?: Record<string, unknown>): Aggregate<Array<R>>;
+    aggregate<R = any>(pipeline: any[], options?: Record<string, unknown>, cb: Function): Promise<Array<R>>;
```

Comment:
I'm pretty sure this will cause a TS error because a mandatory param can't follow an optional param. We'd need to make `options` required here, and add a separate overload for `aggregate(pipeline, cb)`

Reply:
Wow I can't believe I actually didn't notice this lol